src/main/java/org/ow2/proactive/procci/model/occi/platform/bigdata/Swarm.java

```diff
@@ -0,0 +1,146 @@
+            Optional<String> title,
+            List<Mixin> mixins,
+            Optional<String> summary,
+            List<Link> links,
+            Optional<Status> status,
+            Optional<String> machineName,
+            String hostIp,
+            String masterIp,
+            List<String> agentsIp,
+            Optional<String> networkName
+    ) {
+        super(url, kind, title, mixins, summary, links, status);
+        this.machineName = machineName;
+        this.hostIp = hostIp;
+        this.masterIp = masterIp;
+        this.agentsIp = agentsIp;
+        this.networkName = networkName;
+    }
+
+    public static Set<Attribute> createAttributeSet() {
+        Set<Attribute> attributeSet = new HashSet<>();
+        attributeSet.addAll(Component.createAttributeSet());
+        attributeSet.add(MACHINE_NAME);
+        attributeSet.add(HOST_IP);
+        attributeSet.add(MASTER_IP);
+        attributeSet.add(AGENTS_IP);
+        attributeSet.add(NETWORK_NAME);
+        return attributeSet;
+    }
+
+    /**
+     * Convert OCCI swarm to Proactive Cloud Automation Compute
+     *
+     * @param actionType is the action to apply on the swarm
+     * @return the proactive cloud automation model for the swarm
+     */
+    public Model toCloudAutomationModel(String actionType) {
+
+        Model.Builder serviceBuilder = new Model.Builder(SWARM_MODEL, actionType)
+                .addVariable(HOST_IP_NAME, this.getHostIp())
+                .addVariable(MASTER_IP_NAME, this.getMasterIp())
+                .addVariable(AGENTS_IP_NAME, getAgentsIpAsString());
+
+        this.machineName.ifPresent(machineName -> serviceBuilder.addVariable(MACHINE_NAME_NAME, machineName));
+        this.networkName.ifPresent(networkName -> serviceBuilder.addVariable(NETWORK_NAME_NAME, networkName));
+        this.getStatus().ifPresent(
+                status -> serviceBuilder.addVariable(PlatformAttributes.STATUS_NAME, status));
+        this.getTitle().ifPresent(title -> serviceBuilder.addVariable(Attributes.ENTITY_TITLE_NAME, title));
+        this.getSummary().ifPresent(summary -> serviceBuilder.addVariable(Attributes.SUMMARY_NAME, summary));
+
+        this.getMixins().forEach(mixin -> mixin.toCloudAutomationModel(serviceBuilder));
+
+        return serviceBuilder.build();
+    }
+
+    /**
+     * Concatenate the ip contained in the agents ip list
+     *
+     * @return the ip separated by the AGENT_IP_SEPARATOR
+     */
+    private String getAgentsIpAsString() {
+        String agentsIp;
+        if (this.agentsIp.isEmpty()) {
+            return "";
+        }
+        agentsIp = this.agentsIp.get(0);
+        for (int i = 1; i < this.agentsIp.size(); i++) {
+            agentsIp += BigDataIdentifiers.AGENT_IP_SEPARATOR + this.agentsIp.get(i);
+        }
+        return agentsIp;
+    }
+
+    /**
+     * Give the OCCI rendering of a swarm
+     *
+     * @return the swarm rendering
+     */
+    public ResourceRendering getRendering() {
```

Comment:
Here as well you could decouple the logic from the data, by having rendering service classes.